app/models/album.rb

```diff
@@ -2,9 +2,13 @@ class Album < ApplicationRecord
   belongs_to :post
   has_many_attached :images
 
-
+  validates :post, uniqueness: true
```

Comment:
Rails can't provide concurency resitent uniqness validation.
We should add a database constraint (like an index) to be safe.

https://guides.rubyonrails.org/active_record_validations.html#uniqueness